app/validators/answer_code_validator.py

```diff
@@ -48,19 +50,25 @@ def validate_data_version(self):
             self.add_error(self.INCORRECT_DATA_VERSION_FOR_ANSWER_CODES)
 
     def validate_duplicate_answer_codes(self):
-        duplicate_answer_ids = [
-            answer_code["answer_id"]
-            for answer_code in self.answer_codes
-            if "answer_value" not in answer_code
-        ]
-        if duplicates := {
-            answer_id
-            for answer_id in duplicate_answer_ids
-            if duplicate_answer_ids.count(answer_id) > 1
-        }:
-            self.add_error(
-                self.DUPLICATE_ANSWER_CODE_FOUND, duplicate_answer_ids=duplicates
-            )
+        duplicates = find_duplicates(self.codes)
+
+        if len(duplicates) > 0:
+            self.add_error(self.DUPLICATE_ANSWER_CODE_FOUND, duplicates=duplicates)
+        else:
+            duplicate_answer_ids = [
+                answer_code["answer_id"]
+                for answer_code in self.answer_codes
+                if "answer_value" not in answer_code
+            ]
+            if duplicate_ids := {
+                answer_id
+                for answer_id in duplicate_answer_ids
+                if duplicate_answer_ids.count(answer_id) > 1
+            }:
+                self.add_error(
+                    self.DUPLICATE_ANSWER_ID_FOUND,
+                    duplicate_answer_ids=list(duplicate_ids),
```

Comment:
To be consistent with the code, it would call this key `duplicates`.

Reply:
Updated 👍